pkg/meta/base.go

```diff
@@ -948,22 +950,47 @@ func (m *baseMeta) nextInode() (Ino, error) {
 	m.freeMu.Lock()
 	defer m.freeMu.Unlock()
 	if m.freeInodes.next >= m.freeInodes.maxid {
-		v, err := m.en.incrCounter("nextInode", inodeBatch)
-		if err != nil {
-			return 0, err
+		m.prefetchMu.Lock() // Wait until prefetchInodes() is done
+		nextLimit := m.prefetchedInode
+		m.prefetchMu.Unlock()
+		if nextLimit <= m.freeInodes.maxid {
+			v, err := m.en.incrCounter("nextInode", inodeBatch)
+			if err != nil {
+				return 0, err
+			}
+			nextLimit = uint64(v)
+		} else {
+			logger.Debugf("Prefetch hit, next limit: %d", nextLimit)
 		}
-		m.freeInodes.next = uint64(v) - inodeBatch
-		m.freeInodes.maxid = uint64(v)
+		m.freeInodes.next = nextLimit - inodeBatch
+		m.freeInodes.maxid = nextLimit
 	}
 	n := m.freeInodes.next
 	m.freeInodes.next++
 	for n <= 1 {
 		n = m.freeInodes.next
 		m.freeInodes.next++
 	}
+	if m.freeInodes.maxid-m.freeInodes.next < uint64(utils.JitterIt(inodeBatch*0.1)) {
```

Comment:
it will start single goroutine if we use a fixed number here

Reply:
A goroutine will be started when `freeInodes.next` is close to `freeInodes.maxid`, regardless of the threshold.
With a fixed number, when one inode-allocation spawns a goroutine, the next call will always spawn another one. Because the next call will have a larger `freeInodes.next`.
Those goroutines except the first one are unschedulable, they will not consume much resources.

Reply:
```
if m.freeInodes.maxid-m.freeInodes.next  == fixed_one {
}
```

Reply:
Got it, done